hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaDeploymentManagement.java

```diff
@@ -286,7 +284,9 @@ private DistributionSetAssignmentResult assignDistributionSetToTargets(@NotNull
     private void sendDistributionSetAssignmentEvent(final List<JpaTarget> targets, final Set<Long> targetIdsCancellList,
             final Map<String, JpaAction> targetIdsToActions) {
         targets.stream().filter(t -> !!!targetIdsCancellList.contains(t.getId()))
-                .forEach(t -> assignDistributionSetEvent(targetIdsToActions.get(t.getControllerId())));
+                .forEach(t -> sendTargetAssignDistributionSetEvent(targetIdsToActions.get(t.getControllerId())));
```

Comment:
Do you think it's not better to iterate only once over the collection? Additional it's more readable, or?
```
        targets.forEach(target->{
            sendTargetUpdatedEvent(target);
            if(!targetIdsCancellList.contains(target.getId()){
                return;
            }
            
            sendTargetAssignDistributionSetEvent(targetIdsToActions.get(target.getControllerId());
        });
```